libsignal-service/src/provisioning/manager.rs

```diff
@@ -127,10 +84,15 @@ impl<P: PushService> LinkingManager<P> {
         }
     }
 
-    pub async fn provision_secondary_device<R: rand::Rng + rand::CryptoRng>(
+    pub async fn provision_secondary_device<
+        S: PreKeysStore,
+        R: rand::Rng + rand::CryptoRng,
+    >(
         &mut self,
+        aci_store: &mut S,
+        pni_store: &mut S,
         csprng: &mut R,
-        signaling_key: SignalingKey,
+        device_name: &str,
         mut tx: Sender<SecondaryDeviceProvisioning>,
```

Comment:
I would simplify this API. Just take a closure: `FnOnce(Url) -> Option<()>`.

Reply:
Also the `while let` loop below is misleading. It can be replaced by two calls to `next` and expecting url in the first call and message in the second call.

Reply:
Removed the loop in 097bb92